### PR TITLE
#166790056 Pagination support for articles.

### PR DESCRIPTION
**Pull request by shaluchandwani**
#### What does this PR do?
This PR Supports Articles pagination

#### Description of Task to be completed?
Pagination support for articles in 
- HomePage
- profile page

#### How should this be manually tested?
This can be tested by cloning the repo, run the command npm run dev, and then check at the bottom of Home Page and profile page.

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/166790056
	
#### Screenshots (if appropriate)
Cg2yCXuyJo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,4 +6,41 @@ Authors Haven - A Social Platform For The Creative At Heart
...
+## Install dependencies <br/>
+
+```
+npm install
+```
+
+## Starting development server <br/>
+
+```
+npm run start
```
Comment (Joe-Joseph) [inline README.md]: We are using npm run dev to start the server not npm start

File: src/components/Profile/ProfileUserDetails/ProfileUserDetails.js [hunk 2]
```diff
@@ -20,14 +20,15 @@ import { getDataThunk, getDataThunkPrivate, postDataThunkPrivate } from '../../.
 import { getFollowerActionNumber, getFollowingActionNumber } from '../../../redux/actions/followerAction';
 import {
   getUserAction,
-  getAUthorArticlesAction,
+  // getAUthorArticlesAction,
```
Comment (Joe-Joseph) [inline src/components/Profile/ProfileUserDetails/ProfileUserDetails.js]: Please remove the commented line

**Review by Joe-Joseph — APPROVED** [review]

**Review by kagabof — APPROVED** [review]

**Comment by billkabanga** [comment]
Great work on this PR @shaluchandwani, I can't see the story from the PT board attached to this PR.

File: src/components/article/PaginationProfile.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+  paginateArticles = async ({ offset, limit, label }) => {
+    this.setState((prevState) => ({
+      ...prevState,
+      pageNumber: label,
+    }));
+    const id = localStorage.getItem('id');
+    const userName = localStorage.getItem('username');
+    await this.props.getDataThunk('get', `user/${userName}`, getUserAction);
+    await this.props.postDataThunkPrivate('get', `articles/author/${id}?offset=${offset || 0}&limit=${limit || 12}`, getAUthorArticlesAction, null);
```
Comment (billkabanga) [inline src/components/article/PaginationProfile.js]: For this function, Please destructure the props object to access the 2 middleware functions.

**Pull request merged**